### ISSUE TITLE
doc_paragraphs_missing_punctuation doesn't like sentences with bullet points

### ISSUE THREAD

**Issue by TomFryersMidsummer** [rust-lang/rust-clippy#16370]
### Summary

The `clippy::doc_paragraphs_missing_punctuation` doesn't recognise sentences that conclude in a list, numbered or not.

### Lint Name

doc_paragraphs_missing_punctuation

### Reproducer

I tried this code:

```rust
//! This program prints the number 28.
//!
//! It has several security issues, most notably that
//! * it prints your password if your password is 28,
//! * it prints your private RSA key if your private RSA key is 28 and
//! * it prints your mother's maiden name if she indeed used to be Miss 28.

fn main() {
    println!("28")
}
```

I saw this happen:

```
warning: doc paragraphs should end with a terminal punctuation mark
 --> a.rs:3:51
  |
3 | //! It has several security issues, including that
  |                                                   ^ help: end the paragraph with some punctuation: `.`
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#doc_paragraphs_missing_punctuation
  = note: requested on the command line with `-W clippy::doc-paragraphs-missing-punctuation`

warning: 1 warning emitted
```

I expected to see this happen:

[no error]

### Version

```text
rustc 1.94.0-nightly (31cd367b9 2026-01-08)
binary: rustc
commit-hash: 31cd367b9ca1ce359268e7adf4ea540408c0ad85
commit-date: 2026-01-08
host: x86_64-unknown-linux-gnu
release: 1.94.0-nightly
LLVM version: 21.1.8
```

**Comment by relaxcn** [comment]
Maybe you could add a ':' at the end of the line.

```rust
//! It has several security issues, most notably that:
//! * it prints your password if your password is 28,
```

**Comment by TomFryersMidsummer** [comment]
I wouldn't use a colon there, because the preceding words do not form an independent clause.

I could reword the sentence. Indeed, when I ran this lint on our code-base, it flagged two cases like this and I decided one of them would be better reworded. However, I think this lint would be less imposing if it didn't forbid mid-sentence lists.

**Comment by AudaciousAxiom** [comment]
@TomFryersMidsummer I've opened #16487 to hopefully fix this FP.
